src/System.Device.Gpio/System/Device/Analog/AnalogController.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System.Collections.Generic;
+using System.Device.Gpio;
+
+namespace System.Device.Analog
+{
+    public sealed class AnalogController : IDisposable
+    {
+        private readonly AnalogControllerDriver _driver;
+        private readonly HashSet<int> _openPins;
+
+        public event ValueChangeEventHandler ValueChanged;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GpioController"/> class that will use the specified numbering scheme and driver.
+        /// </summary>
+        /// <param name="numberingScheme">The numbering scheme used to represent pins provided by the controller.</param>
+        /// <param name="driver">The driver that manages all of the pin operations for the controller.</param>
+        public AnalogController(PinNumberingScheme numberingScheme, AnalogControllerDriver driver)
```

Comment:
This whole PR is also a preview/suggestion for #878 - so that's intentional. 
I'm not sure this should inherit from GpioController, since typically you can't change the pin modes and you can't write to the pins. So an implementation would typically have to throw lots of "NotSupportedExceptions" which is not so nice.

Reply:
Writing to an analogic pin is placing the pin into a PWM mode. And you can always change the pin mode (up to the fact that a pin supports or not a specific analogic/digital/pul)

Reply:
That's true. Firmata actually names PWM analog out (although technically that's something else, and the Arduino Due for instance has real analog out pins). But mixing PWM and analog in gets a bit confusing for boards that don't support either. 

I'm not sure what the best approach is here, but right now it looks like keeping analog in (and eventually out) separate from PWM and GPIO would keep the interface simpler and reduce the possibilities of a wrong API usage.

Reply:
Fair. so debate still open for the *"Analog Controller"* or equivalent naming and link to anything existing.